File: Yandex/Practicum/Algorightms_and_Data_Structures/8_sprint/A._Разворот_строки/solution.cpp
#include <iostream>
#include <vector>
#include <string>


void solution() {
    /*
    Time Complexity O(N)
    Memory Complexity O(1)
    */
    std::vector<std::string> words;
    std::string word = "";
    while (std::cin >> word) {
        words.push_back(word);
    }
    int left = 0;
    int right = words.size() - 1;
    while (left < right) {
        std::swap(words[left], words[right]);
        ++left;
        --right;
    }
    for (size_t i = 0; i < words.size(); ++i) {
        if (i) std::printf(" ");
        std::cout << words[i];
    }
    std::printf("\n");
}


int main() {
    solution();
}

File: Yandex/Practicum/Algorightms_and_Data_Structures/8_sprint/A._Разворот_строки/solution.py
import sys


def solution() -> None:
    """
    Time Complexity O(N)
    Memory Complexity O(1)
    """
    words: list[str] = sys.stdin.readline().rstrip().split()
    left: int = 0
    right: int = len(words) - 1
    while left < right:
        words[left], words[right] = words[right], words[left]
        left += 1
        right -= 1
    sys.stdout.write(' '.join(words) + '\n')


if __name__ == '__main__':
    solution()